common/static/common/js/discussion/utils.js

```diff
@@ -444,19 +463,52 @@
             }
         };
 
-        DiscussionUtil.abbreviateHTML = function(html, minLength) {
+        DiscussionUtil.convertMath = function(element, alreadyEscaped) {
+            var elementContent = element.html();
+            if (alreadyEscaped){
+                // markdownWithHighlight requires unescaped text, so we must unescape it.
+                // safe-lint: disable=javascript-escape
+                elementContent = _.unescape(edx.HtmlUtils.HTML(elementContent));
+            }
+            edx.HtmlUtils.setHtml(
+                element,
+                this.postMathJaxProcessor(this.markdownWithHighlight(elementContent))
+            );
+        };
+
+        DiscussionUtil.abbreviateHTML = function(htmlSnippet, maxLength) {
             var $result, imagesToReplace, truncated_text;
-            truncated_text = jQuery.truncate(html, {
-                length: minLength,
+            truncated_text = edx.HtmlUtils.HTML(jQuery.truncate(htmlSnippet.toString(), {
+                length: maxLength,
                 noBreaks: true,
                 ellipsis: gettext('…')
-            });
-            $result = $("<div>" + truncated_text + "</div>");
+            }));
+            $result = $(edx.HtmlUtils.joinHtml(
+                edx.HtmlUtils.HTML("<div>"),
+                truncated_text,
+                edx.HtmlUtils.HTML("</div>")
+            ).toString());
             imagesToReplace = $result.find("img:not(:first)");
             if (imagesToReplace.length > 0) {
-                $result.append("<p><em>Some images in this post have been omitted</em></p>");
+                edx.HtmlUtils.append(
+                    $result,
+                    edx.HtmlUtils.interpolateHtml(
+                        edx.HtmlUtils.HTML("<p><em>{text}</em></p>"),
+                        {text: gettext("Some images in this post have been omitted")}
+                    )
+                );
             }
-            imagesToReplace.replaceWith("<em>image omitted</em>");
+            // has to do htmlSnippet + ensureHtml because linter still reports javascript-jquery-insertion
+            // in the following (valid?) cases:
+            // 1. imagesToReplace.after(afterMessage.toString())
+            // 2. imagesToReplace.after(edx.HtmlUtils.interpolateHtml(
+            //     edx.HtmlUtils.HTML("<em>{text}</em>"), {text: gettext("image omitted")}
+            // ).toString())
+            // It does not report if it is written on single line, but than it becomes line too long jshint violation
```

Comment:
@e-kolpakov: My opinion is that this is helpful, but verbose and not the best place for this.  We'd need the same verbose comment every place we use ensureHtml() to ensure we have an HTML Snippet.

I added the following JIRA ticket: https://openedx.atlassian.net/browse/TNL-4983

My recommendation would be to either not have anything in code (just in the Read the Docs), or to add a one-liner above 511 as follows:
```
// See TNL-4983 for an explanation of why the linter requires ensureHtml()
```

Let me know what you think of this proposal.